packages/aws-cdk/test/api/cloudformation-deployments.test.ts

```diff
@@ -1,18 +1,38 @@
 jest.mock('../../lib/api/deploy-stack');
 
+import { CloudFormation } from 'aws-sdk';
 import { CloudFormationDeployments } from '../../lib/api/cloudformation-deployments';
 import { deployStack } from '../../lib/api/deploy-stack';
 import { ToolkitInfo } from '../../lib/api/toolkit-info';
+import { CloudFormationStack } from '../../lib/api/util/cloudformation';
 import { testStack } from '../util';
 import { mockBootstrapStack, MockSdkProvider } from '../util/mock-sdk';
+import { FakeCloudformationStack } from './fake-cloudformation-stack';
 
 let sdkProvider: MockSdkProvider;
 let deployments: CloudFormationDeployments;
 let mockToolkitInfoLookup: jest.Mock;
+let currentCfnStackResources: { [key: string]: CloudFormation.StackResourceSummary[] };
+let numberOfTimesListStackResourcesWasCalled: number;
 beforeEach(() => {
   jest.resetAllMocks();
   sdkProvider = new MockSdkProvider();
   deployments = new CloudFormationDeployments({ sdkProvider });
+
+  numberOfTimesListStackResourcesWasCalled = 0;
+  currentCfnStackResources = {};
+  sdkProvider.stubCloudFormation({
+    listStackResources: ({ StackName: stackName }) => {
+      numberOfTimesListStackResourcesWasCalled++;
+      if (!currentCfnStackResources[stackName]) {
```

Comment:
Can we save the `currentCfnStackResources[stackName]` expression in a local variable? This way, we won't have to repeat it below.